### PR TITLE
[CALCITE-3733] MySQL JDBC rewrite should rewrite CAST as TIMESTAMP to CAST as DATETIME (Vineet Garg)

### PR DESCRIPTION
**Pull request by vineetgarg02**

### CODE REVIEW

**Comment by DonnyZone** [comment]
I noticed that the timestamp type is supported in MySQL (https://dev.mysql.com/doc/refman/8.0/en/datetime.html).
Is the `cast (x as timestamp)` not supported?

**Comment by vineetgarg02** [comment]
> I noticed that the timestamp type is supported in MySQL (https://dev.mysql.com/doc/refman/8.0/en/datetime.html).
> Is the `cast (x as timestamp)` not supported?

@DonnyZone Yes `cast(x as timestamp)` is not supported in MySQL

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -1834,6 +1834,18 @@ private static String toSql(RelNode root, SqlDialect dialect) {
     sql(query).withMysql().ok(expected);
   }
 
+  @Test void testMySqlCastToTimestamp() {
+    //final String query = "select cast(\"hire_date\" as TIMESTAMP), \"employee_id\" "
+     //   + "from \"employee\" ";
+    //final String expected = "select cast(\"hire_date\" as DATE_TIME), \"employee_id\" "
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:1840]: Remove the comment

**Comment by danny0405** [comment]
Thanks for the fix @vineetgarg02 , let's give this patch a better commit message:
"In JDBC adapter, when generating SQL for MYSQL, generate TIMESTAMP type as DATETIME for CAST"

**Comment by vineetgarg02** [comment]
> Thanks for the fix @vineetgarg02 , let's give this patch a better commit message:
> "In JDBC adapter, when generating SQL for MYSQL, generate TIMESTAMP type as DATETIME for CAST"

@danny0405  Thanks for taking a look, i have addressed your comments/concerns.

**Pull request merged**